agent/engine/docker_task_engine.go

```diff
@@ -538,13 +538,19 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 			return engine.client.ImportLocalEmptyVolumeImage()
 		}
 	}
+
+	// Record the pullStoppedAt timestamp
```

Comment:
What happens if this doesn't happen?  What happens if the agent crashes right before this code executes?

Reply:
If this doesn't happen, this field will be empty and the task won't have `pullstoppedat` timestamps. Are you concerning about container moved to stopped without pulling?  If agent crashes here, agent should try to pull the image again when it start, then it will be set by that time.